### ISSUE TITLE
Change Traffic Ops Server Unique Key from IP,Profile to IP,Port

### ISSUE THREAD

**Issue by rob05c** [apache/trafficcontrol#4940]
IP+Port makes much more sense for the Unique Key. The IP+Port is what uniquely defines the server for requests (At the HTTP level, arguably FQDN+Port, but the current key is IP, which is the unique identifier at the IP level).

The current IP,Profile setting is undesirable because it makes creating multiple similar servers a huge pain, having to create many duplicate profiles. It also allows for duplicate IP+Port, which should not be allowed. TC should be standardized around either IP+Port or FQDN+Port (possibly both).

Labeling 'enhancement' - but this is a on the line of bug, enhancement, and tech debt.

## I'm submitting a ...
-  improvement request (usability, performance, tech debt, etc.)

## Traffic Control components affected ...
-  Traffic Ops

## Current behavior:
TO Server Unique key is IP+Profile

## Expected / new behavior:
TO Server Unique key should be IP+Port. Consider also adding a Domain+Hostname+Port Unique Key.

## Minimal reproduction of the problem with instructions:
Try to make 2 servers with the same IP and Profile and different Ports. Observe error.

## Anything else:

**Comment by jhg03a** [comment]
This is a problem for MSO VIP.  We can make additional MSO server profiles to satisfy the unique constraint.  We can't force load balancers to use multiple ports.

**Comment by ocket8888** [comment]
Somebody needs to refresh me on why MSO uses servers instead of Origins.

**Comment by jhg03a** [comment]
Servers came first and origins is backward compatible with it.

**Comment by jhg03a** [comment]
The main feature gain Origins has that servers doesn't is the ability to assign a geo-coordinate to an individual origin instead of an origin cachegroup like in the past.

**Comment by ocket8888** [comment]
I feel like there was a reason we couldn't switch to Origins yet, though, and idk if/where that discussion was captured.

**Comment by mitchell852** [comment]
is this still relevant? did any of this change with the introduction of server interfaces?

**Comment by ocket8888** [comment]
No, with interfaces, the IP the issue refers to is now the "service IP address(es)" of the cache server. It's become a much more complex key, but not less reasonable, really.

**Comment by mitchell852** [comment]
> No, with interfaces, the IP the issue refers to is now the "service IP address(es)" of the cache server. It's become a much more complex key, but not less reasonable, really.

^^ it sounds like that answer is a "yes, problem still exists". is that true?

so can i do this currently?

server 1: profile A, IP address=1.1.1.1, port=80
server 2: profile A, IP address=1.1.1.1, port=8080

**Comment by bdgranger** [comment]
I haven't tried it recently, but I think that the uniqueness constraint
will still flag this as not allowed because the profile is the same?

**Comment by ocket8888** [comment]
Yeah, the constraint is still profile and IP afaik
